### PR TITLE
fix: Geolocator.RequestAccessAsync() not returning on ios14

### LINKED ISSUE #4853: Geolocator.RequestAccessAsync() Hangs on IOS 14.x
## Current behavior

Geolocator.RequestAccessAsync() never returns if the location permission is denied - in iOS 14 and later only.

## Expected behavior

The function should return null or throw an exception.

## How to reproduce it (as minimally and precisely as possible)

Run this:

                var accessStatus = await Geolocator.RequestAccessAsync();

                if (accessStatus == GeolocationAccessStatus.Allowed)
                   {
                    Geolocator geoLocator = new Geolocator();
                    Geoposition pos = await geoLocator.GetGeopositionAsync();
                   }

## Workaround

I don't have one, other than directing thousands of users to manually grant location permissions to my app.

## Environment

Nuget Package Version(s):

[x] Uno.Core v2.1.0
[x] Uno.Microsoft.Toolkit.Uwp v6.1.0-build.205.g2e0e348372
[x] Uno.UI v3.5.0-dev.57
[x] Uno.UI.RemoteControl v3.5.0-dev.57
[x] Uno.UniversalImageLoader v1.9.33

Affected platform(s):

[ X] iOS

IDE:

- [X] Visual Studio 2019 (version: )

Relevant plugins:

None.

## Anything else we need to know?

No.

### PR DESCRIPTION
**Pull request by Xiaoy312**
GitHub Issue (If applicable): resolved #4853

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

The `Geolocator.RequestAccessAsync()` method never returns after permission is denied or granted, be promptless or after the user has responded to the permission prompt.

## What is the new behavior?

The `Geolocator.RequestAccessAsync()` method should return as soon as the location permission is updated.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information
The issue was with `CLLocationManager.AuthorizationChanged` never being called. We had to use the same/similar method from `CLLocationManagerDelegate` which actually works.

### CODE REVIEW

**Review by MartinZikmund — APPROVED** [review]
👍👍👍

**Review by ajpinedam — APPROVED** [review]

File: src/Uno.UWP/Devices/Geolocation/Geolocator.iOSmacOS.cs [hunk 1]
```diff
@@ -122,15 +123,21 @@ private static async Task<GeolocationAccessStatus> RequestAccessInternalAsync()
 
 				try
 				{
-					GeolocationAccessStatus accessStatus;
-					var cts = new TaskCompletionSource<CLAuthorizationStatus>();
+					var accessStatus = default(GeolocationAccessStatus);
+					var tsc = new TaskCompletionSource<CLAuthorizationStatus>();
 
+#if __IOS__
+					var @delegate = new CLLocationManagerDelegate();
```
Comment (davidjohnoliver) [inline src/Uno.UWP/Devices/Geolocation/Geolocator.iOSmacOS.cs:130]:
```suggestion
					// Workaround for a bug in Xamarin.iOS https://github.com/unoplatform/uno/issues/4853
					var @delegate = new CLLocationManagerDelegate();

```

**Review by davidjohnoliver — APPROVED** [review]
You should add a manual test that QA can run: https://platform.uno/docs/articles/uno-development/working-with-the-samples-apps.html#adding-a-manual-test-sample

Also, remember to use the GitHub automatic close keywords ( `fixes #4853`) in the PR description.

**Comment by Xiaoy312** [comment]
There is an existing manual test: 
```

**Comment by Xiaoy312** [comment]
`

**Comment by davidjohnoliver** [comment]
> There is an existing manual test:
> 
> ```
> [SampleControlInfo("Windows.Devices", "Geolocator", description: "Demonstrates use of Windows.Devices.Geolocation.Geolocator", viewModelType: typeof(GeolocatorTestsViewModel))]
> ```

Good, just make sure there's a clear set of instructions to repro the bug placed on the page, and tag the attribute `IsManualTest` as mentioned in the docs I linked.

**Pull request merged**